### ISSUE TITLE
A suggestion: add an option to all built-in loss functions to avoid summation

### ISSUE THREAD

**Issue by xuancong84** [pytorch/pytorch#1657]
I would like to do instance re-weighting, so every instance in a minibatch has its own instance weight. However, current implementation of all loss functions return the single summed-up loss variable. Thus, it is not possible to do this without re-implementing the loss function myself, doing which will lose speed optimization from using built-in loss functions.

My suggestion is that for every loss function, add an option (say _return_raw_values_=True/False), setting it to true will cause the loss function to return the loss tensor variable before the summation.

This should be quite straight-forward to implement. Thanks!

**Comment by fmassa** [comment]
This is being tracked in https://github.com/pytorch/pytorch/issues/563

**Comment by soumith** [comment]
because this is already treacked in #563 i'll close this. thanks.
